### ISSUE TITLE
Move "missing await" analyzer from vs-threading to a common place

### ISSUE THREAD

**Issue by jasonmalinowski** [dotnet/roslyn-analyzers#2644]
The compiler warns if you call a Task-returning method in an async method that you may have forgotten the "await" on it. It _doesn't_ do this in a regular method -- there it's just a fire-and-forget behavior and there's nothing to hint otherwise. @NickCraver was looking for an analyzer to do this but it turns out we don't really have one in the box...we should have one.

It turns out the vs-threading repo already has [exactly this analyzer](https://github.com/microsoft/vs-threading/blob/master/doc/analyzers/VSTHRD110.md) -- we should consider moving that to some core analyzer package here. In @NickCraver's case we could suggest to add the vs-threading analyzers package as a way to run this analyzer, but he's not writing a VS extension so many of the other analyzers wouldn't be giving good advice, so that's not a sustainable suggestion.

**Comment by mavasani** [comment]
I believe @AArnott has proposed adding a package dependency from FxCopAnalyzers to VS Threading analyzers so all the clients automatically get the threading analyzers.

**Comment by AArnott** [comment]
> but he's not writing a VS extension so many of the other analyzers wouldn't be giving good advice

@jasonmalinowski  Which analyzers in vs-threading would give bad advice outside of VS? We've written them all to not be VS specific.
Some do not apply to a *web* application, but they mostly self-deactivate where possible, and users can turn a rule off if it's inappropriate.

**Comment by jasonmalinowski** [comment]
I all the rules about anything involving the UI thread wouldn't apply due to lack of a UI thread. But if those analyzers are written to self-disable and that's a scenario you actually want to *support* then I could get behind that plan. I was (apparently incorrectly) assuming that installing that package outside of VS scenarios simply wasn't something you wanted in your test matrix.

**Comment by AArnott** [comment]
Yes, we intend for this to be used outside VS. The level of support for them probably (officially) drops once it's a non-VS scenario. But I'm certainly interested in continuing to support it as we have for the last several years. We have plenty of folks already using it and have had a little feedback to help us make it work better outside of VS scenarios.

**Comment by sharwell** [comment]
I like the idea of the vs-threading analyzers being a broader set of threading-related analyzers which work with or without vs-threading/JTF types. It seems viable to add the package reference so it gets picked up automatically with the code analysis rules here, but we have some discussion to do relating to the generated ruleset files we distribute.

**Comment by sharwell** [comment]
Reported again at AB#1749088
